src/Raven.Studio/typescript/common/shell/accessManager.ts

```diff
@@ -113,19 +113,27 @@ class accessManager {
             case "Operator":
                 return this.isOperatorOrAbove();
         }
-        
+
         const activeDatabaseAccess = this.activeDatabaseAccessLevel();
 
         if (!activeDatabaseAccess) {
             return false;
         }
 
-        if (activeDatabaseAccess === 'Read' &&
-            (requiredAccess === "DatabaseReadWrite" || requiredAccess === "DatabaseAdmin")) {
+        return this.canHandleOperationDatabaseLevelCheck(activeDatabaseAccess, requiredAccess as databaseAccessLevel);
+    }
+    
+    canHandleOperationDatabaseLevelCheck(actualLevel: databaseAccessLevel, requiredLevel: accessLevel) {
```

Comment:
do we need separeate function?

Reply:
I separated this because it is called from 2 places:
1. from `canHandleOperation` (in accessManager.ts)
2. from `allowToAccessView` (in viewModelBase.ts Line #83)

Reply:
ok but I'd like to avoid having separate functions per global and per database. 

I think we should have single function: 

canHandleOperation(level: accessLevel) 

and then we have all magic inside. 

Optionally: we could pass **optional** database name. 

So those function should work for **All** types not only for database level. 

Unless I'm missing something.

Reply:
done